pkg/appmanager/appManager.go

```diff
@@ -1280,16 +1281,20 @@ func (appMgr *Manager) syncRoutes(
 				}
 				if dep.Kind == RuleDep {
 					for _, pol := range rsCfg.Policies {
+						var toRemove []*Rule
 						for _, rl := range pol.Rules {
 							if rl.FullURI == dep.Name {
-								rsCfg.DeleteRuleFromPolicy(pol.Name, rl, appMgr.mergedRulesMap)
+								toRemove = append(toRemove, rl)
 								// Delete profile (route only)
 								if rsCfg.MetaData.ResourceType == "route" {
 									resourceName := strings.Split(rl.Name, "_")[3]
 									rsCfg.DeleteRouteProfile(dep.Namespace, resourceName)
 								}
 							}
 						}
+						for _, rl := range toRemove {
+							rsCfg.DeleteRuleFromPolicy(pol.Name, rl, appMgr.mergedRulesMap)
```

Comment:
Does this have to be removed after looping over the policy rules? It can't be removed inline?